### PR TITLE
fix: [CDS-39785]: cypress tests added for template references

### PR DESCRIPTION
**Pull request by samarth-sharma31**
#### Screenshots

Screenshot 2022-07-13 at 3 09 29 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
retrigger titlecheck

**Comment by diptiman-adak** [comment]
retrigger sonar

**Comment by diptiman-adak** [comment]
retrigger sonar

File: cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { templatesListRoute, gitSyncEnabledCall } from '../../support/70-pipeline/constants'
+
+describe('Template Reference By', () => {
+
+    const templatesListCall =
```
Comment (depanshu-dhiman) [inline cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts:12]: Move to constants

File: cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    beforeEach(() => {
+        cy.on('uncaught:exception', () => {
+            // returning false here prevents Cypress from
+            // failing the test
+            return false
+        })
+        cy.intercept('GET', gitSyncEnabledCall, { connectivityMode: null, gitSyncEnabled: false })
+        cy.intercept('POST', templatesListCall, { fixture: 'template/api/templatesList' }).as('templatesListCall')
+        cy.intercept('POST', '/template/api/templates/list?routingId=accountId&accountIdentifier=px7xd_BFRCi-pfWPYXVjvw&orgIdentifier=default&projectIdentifier=Kapil&templateListType=All&module=cd', { fixture: 'template/api/templatesList' }).as('templatesListCallForDrawer')
```
Comment (depanshu-dhiman) [inline cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts:25]: Extract url in variable

**Review by depanshu-dhiman — CHANGES_REQUESTED** [review]
Minor comments

File: cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts (hunk 1 above)
Comment (samarth-sharma31) [reply cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts]: done

File: cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { templatesListRoute, gitSyncEnabledCall } from '../../support/70-pipeline/constants'
+
+describe('Template Reference By', () => {
+
+    const templatesListCall =
+        '/template/api/templates/list?routingId=accountId&accountIdentifier=accountId&projectIdentifier=project1&orgIdentifier=default&templateListType=LastUpdated&searchTerm=&page=0&sort=lastUpdatedAt%2CDESC&size=20'
+    const templateReferencesCall =
```
[the remark this answers is not in the document]
Comment (samarth-sharma31) [reply cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts]: done

File: cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts (hunk 2 above)
Comment (samarth-sharma31) [reply cypress/integration/72-templates-library/TemplateReferenceBy.spec.ts]: yes, done

**Review by depanshu-dhiman — APPROVED** [review]

**Comment by samarth-sharma31** [comment]
retrigger jest

**Pull request merged**